rir/src/interpreter/interp.cpp

```diff
@@ -346,6 +346,46 @@ static void loopTrampoline(Code* c, InterpreterInstance* ctx, SEXP* env,
     Rf_endcontext(&cntxt);
 }
 
+const static SEXP inlineContextTrampolineDeoptMarker = (SEXP)0x7008;
+static SEXP inlineContextTrampoline(Code* c, SEXP ast, SEXP sysparent, SEXP op,
+                                    InterpreterInstance* ctx, Opcode* pc,
+                                    R_bcstack_t* localsBase) {
+    CallContext call(nullptr, op, -1, ast, 0, nullptr, nullptr, sysparent,
+                     Assumptions(), ctx);
+    RCNTXT cntxt;
+    // The first env should be the callee env, but that will be done by the
+    // callee. We store sysparent there, because our optimizer may actually
+    // delay instructions into the inlinee and might assume that we still have
+    // to outer env.
+    initClosureContext(ast, &cntxt, sysparent, sysparent,
+                       symbol::delayedArglist, op);
+
+    auto trampoline = [&]() {
+        if ((SETJMP(cntxt.cjmpbuf))) {
```

Comment:
stjmp-longjmp is used for non-local returns. You can jump from a nested function to this place. And you can also restart a function instead of just returning (used for continuations)